Desktop/widgets/listmodellabelvalueterms.cpp

```diff
@@ -132,23 +116,25 @@ void ListModelLabelValueTerms::_setLabelValues(const JASPListControl::LabelValue
 
 void ListModelLabelValueTerms::setLabelValuesFromSource()
 {
-	JASPListControl::LabelValueMap values;
+	JASPListControl::LabelValueMap labelValuePairs;
 
 	if (_listView->addEmptyValue())
-		values.push_back(std::make_pair(_listView->placeholderText(), ""));
+		labelValuePairs.push_back(std::make_pair(_listView->placeholderText(), ""));
 
-	for (const std::pair<JASPListControl::SourceType *, Terms>& source : _listView->getTermsPerSource())
+	for (const auto& pair : listView()->getTermsPerSource())
```

Comment:
Or maybe do it like `void		applyToAll(std::function<void(Analysis *analysis)> applyThis);`
The you could just have it like:
` void forAllSourceTerms(std::function<void(SourceItem * item, const Terms & terms)> callThis);`

That way all of these `for ... copy first and second, do stuff` loops can be so much nicer than what Youve been adding.

Reply:
You could even make one where you check if it has a model and make a `forAllSourceTermsWithModel(std::function<void(SourceItem * item, ListModel * model)> callThis)` or something like that

Reply:
Nice. I'll do this